### ISSUE TITLE
Discussion: New ID types for parcels

### ISSUE THREAD

**Issue by thomastaylor312** [deislabs/bindle#253]
After some great discussion and feedback at Cloud Native Wasm Day (huge hat tip to @carlsverre for this), I think we should have a discussion about how we attach IDs to parcels. There are 2 distinct possibilities here: [CIDs](https://github.com/multiformats/cid) and [DIDs](https://w3c.github.io/did-core/). 

Here are my initial observations as to the benefits/tradeoffs of each one

**CIDs**

- Still use SHA256 for the underlying multihash, but has support for other hash types in the future
- Universally unique ID that can be used with things like IPFS
- Lots of libraries available (including Rust) for different languages
- I don't know if it is super widely used outside of IPFS or not. Basically it isn't a "standard" in the strictest sense

**DIDs**

- A W3C standard
- Identifiers that are generated and assigned by the specific entity
- This approach seems a little heavyweight for what we need in Bindle

Basically I feel like CIDs might be a better option for us, but I want to leave it open for discussion so we can do some more research
